crates/wysiwyg/src/dom/range.rs

```diff
@@ -399,6 +399,22 @@ impl Range {
             .max()
     }
 
+    pub(crate) fn deepest_container_node(
+        &self,
+        ancestor_of: Option<&DomHandle>,
+    ) -> Option<&DomLocation> {
+        self.locations
+            .iter()
+            .filter(|l| {
+                let mut found = true;
+                if let Some(ancestor_of) = ancestor_of {
+                    found = l.node_handle.is_ancestor_of(ancestor_of);
+                }
+                found && !l.is_leaf()
+            })
+            .max()
+    }
+
```

Comment:
Could this be a call to `deepest_node_of_kind` where it adds the container kind to the function call for you?

Reply:
If it was only for formatting we'd still have to rework the `deepest_node_of_kind` func because it just compares `kind` and things like `kind() == DomNodeKind::Formatting(_)` is not allowed.
Moreover we probably want to include any type of container that might be present in between since it could be links, lists, etc and these should update the menu state as well.